drivers/dma/dma_mcux_lpc.c

```diff
@@ -179,12 +182,53 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
```

Comment:
you need to handle the case where the driver is re-configuring the triggering channel to not be linked to the previously linked channel anymore, for which config->linked_channel will be NULL, and at that point you should undo the inputmux connection, reconfigure the dma channel, and give the semaphore. You also need to handle the case where the currently triggering channel wants to link to a different channel, and in this case the driver should know which channel is currently using the Otrig and recognize when it is not needed to take the semaphore to reconfigure that channel

Reply:
I believe this case has been fixed but check just in case.

Reply:
@EmilioCBen still missing the case where one of the channels that is already linked to another is reconfigured to link to a different one, but there is a question of how that is supposed to work in the api imo since it isnt documented. maybe the user would want to link to two different ones, but I think for now actually let's not assume that's how the api works and only allow linking a channel to one other channel and not multiple

Reply:
So currently, if 
   1. A is configured to link with B
   2. A is configured to link with C
Then
   When A is finished with its transfer, both B and C fire off their transfer
Thus, 
   A can chain with multiple channels when "reconfigured", and while this behavior is _NOT_ defined in the api
   this effect _DOES_ represent what the mucx sdk sample implies when using their LPC DMA Channel Chaining Sample.
Therefore, if the user wants a different behavior, we should probably add a specific api function in the driver to unchain one of the otrig muxes.

Reply:
The contract for this driver implementation is defined by the Zephyr subsystem API, not what the mcuxpresso sample does.

Reply:
The Driver no longer allows the channel to be linked with multiple channels, instead it will "reconfigure" the channel and will remove the linkage of the channel or changed what the current channel is linked to, depending on what the user passes in.